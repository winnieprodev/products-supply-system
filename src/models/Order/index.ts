export * from "./enums";
export * from "./Order.d";
