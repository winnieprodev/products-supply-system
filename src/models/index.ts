export * from "./Cart";
export * from "./Order";
export * from "./Outlet";
export * from "./Product";
export * from "./Search";
export * from "./Staff";
export * from "./Supplier";
export * from "./User";
export * from "./utils";
