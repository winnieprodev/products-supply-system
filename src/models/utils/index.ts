export * from "./async.d";
export * from "./enums";
