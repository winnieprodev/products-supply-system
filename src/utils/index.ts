export * from "./routeUtils";
export * from "./supplierUtils";
export * from "./testUtils";
