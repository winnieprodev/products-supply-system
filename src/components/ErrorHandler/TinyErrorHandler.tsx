import { ErrorIcon, Dropdown } from "components";
import { Error, WrapperMode } from "models";
import React from "react";

import * as Styled from "./errorHandler.styled";

interface OwnProps {
  error: Error;
  message?: string;
}

export const TinyErrorHandler: React.FC<OwnProps> = ({
  error,
  message = "Sorry, we've encountered an issue",
}) => {
  return (
    <Dropdown
      anchor={
        <Styled.TinyAnchor>
          <Styled.Icon>
            <ErrorIcon />
          </Styled.Icon>
        </Styled.TinyAnchor>
      }
    >
      <Styled.Wrapper $mode={WrapperMode.tiny}>
        <Styled.Title>Error</Styled.Title>
        <Styled.Message>{message}</Styled.Message>
        <Styled.Description>error details:</Styled.Description>
        <Styled.ErrorMessage>{error.message}</Styled.ErrorMessage>
      </Styled.Wrapper>
    </Dropdown>
  );
};

export default React.memo(TinyErrorHandler);
