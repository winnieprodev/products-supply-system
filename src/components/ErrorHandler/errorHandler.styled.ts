import { WrapperMode } from "models";
import styled, { css } from "styled-components";
import {
  errorBackgroundColor,
  errorBackgroundColorDarker,
  errorColor,
} from "styles/colors";

interface WrapperProps {
  $mode: WrapperMode;
}

export const Wrapper = styled.div<WrapperProps>`
  ${({ $mode }) =>
    $mode === WrapperMode.full
      ? css`
          margin: 10vh auto 0;
        `
      : css`
          margin: 1vh auto;
        `}
  background: ${errorBackgroundColor};
  padding: 16px;
  width: 260px;
  border-radius: 6px;
  color: ${errorColor};
  ${({ $mode }) =>
    $mode === WrapperMode.tiny &&
    css`
      background: transparent;
    `}
`;

export const TinyAnchor = styled.div`
  background: ${errorBackgroundColor};
  border-radius: 50%;
  padding: 4px;
  width: 34px;
  color: ${errorColor};
  margin-left: 4px;
`;

export const Icon = styled.div`
  text-align: center;
  font-size: 24px;
  margin: 4px;
`;

export const Title = styled.div`
  text-align: center;
  font-size: 24px;
  font-weight: bold;
  margin-bottom: 16px;
`;

export const Message = styled.div`
  font-size: 16px;
  margin-bottom: 16px;
  text-align: center;
`;

export const Description = styled.div`
  font-size: 10px;
  font-weight: bold;
  text-align: center;
  margin-bottom: 4px;
`;

export const ErrorMessage = styled.div`
  background: ${errorBackgroundColorDarker};
  margin: 0;
  color: initial;
  padding: 16px 10px;
  font-size: 13px;
  border-radius: 4px;
  font-family: monospace;
`;
