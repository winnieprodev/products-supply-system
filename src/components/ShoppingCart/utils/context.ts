import { Async, Cart, ProductCartInfo } from "models";
import React from "react";
import { mockAsync, mockCart } from "utils";

interface ShoppingCartContextProps {
  cartId: Async<number>;
  cart: Async<Cart> | undefined;
  setCart: (cart: Async<Cart> | undefined) => void;
  setProduct: (product: ProductCartInfo) => void;
  removeProduct: (productId: string) => void;
}

export const EMPTY_CART = { products: [] };
const mockFunction = () => {};

export const ShoppingCartContext = React.createContext<ShoppingCartContextProps>(
  {
    cartId: mockAsync(0),
    cart: mockAsync(mockCart(0)),
    setCart: mockFunction,
    setProduct: mockFunction,
    removeProduct: mockFunction,
  },
);
