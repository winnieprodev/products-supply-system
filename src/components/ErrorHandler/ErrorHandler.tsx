import { ErrorIcon } from "components";
import { Error, WrapperMode } from "models";
import React from "react";

import * as Styled from "./errorHandler.styled";
import TinyErrorHandler from "./TinyErrorHandler";

interface OwnProps {
  error: Error;
  message?: string;
  mode?: WrapperMode;
}

export const ErrorHandler: React.FC<OwnProps> = ({
  error,
  message = "Sorry, we've encountered an issue",
  mode = WrapperMode.full,
}) => {
  console.error(error);
  if (mode === WrapperMode.tiny) {
    return <TinyErrorHandler error={error} message={message} />;
  }
  return (
    <Styled.Wrapper $mode={mode}>
      <Styled.Icon>
        <ErrorIcon />
      </Styled.Icon>
      <Styled.Title>Error</Styled.Title>
      <Styled.Message>{message}</Styled.Message>
      <Styled.Description>error details:</Styled.Description>
      <Styled.ErrorMessage>{error.message}</Styled.ErrorMessage>
    </Styled.Wrapper>
  );
};

export default React.memo(ErrorHandler);
