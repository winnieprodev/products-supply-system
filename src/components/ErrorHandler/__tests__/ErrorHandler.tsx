import { render } from "@testing-library/react";
import { WrapperMode } from "models";
import { BrowserRouter } from "react-router-dom";
import { mockError } from "utils";

import ErrorHandler from "../ErrorHandler";

describe("ErrorHandler", () => {
  describe("Snapshot tests", () => {
    test("it matches the snapshot", () => {
      const { container } = render(
        <ErrorHandler error={mockError("message")} />,
      );
      expect(container.firstChild).toMatchSnapshot();
    });
    test("it matches the snapshot for mode tiny", () => {
      const { container } = render(
        <BrowserRouter>
          <ErrorHandler error={mockError("message")} mode={WrapperMode.tiny} />
        </BrowserRouter>,
      );
      expect(container.firstChild).toMatchSnapshot();
    });
    test("it matches the snapshot for mode singleLine", () => {
      const { container } = render(
        <ErrorHandler
          error={mockError("message")}
          mode={WrapperMode.singleLine}
        />,
      );
      expect(container.firstChild).toMatchSnapshot();
    });
  });
});
