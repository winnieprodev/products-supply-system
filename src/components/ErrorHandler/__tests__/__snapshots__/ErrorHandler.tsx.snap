// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ErrorHandler Snapshot tests it matches the snapshot 1`] = `
.c0 {
  margin: 10vh auto 0;
  background: #FFECEC;
  padding: 16px;
  width: 260px;
  border-radius: 6px;
  color: #9C2828;
}

.c1 {
  text-align: center;
  font-size: 24px;
  margin: 4px;
}

.c2 {
  text-align: center;
  font-size: 24px;
  font-weight: bold;
  margin-bottom: 16px;
}

.c3 {
  font-size: 16px;
  margin-bottom: 16px;
  text-align: center;
}

.c4 {
  font-size: 10px;
  font-weight: bold;
  text-align: center;
  margin-bottom: 4px;
}

.c5 {
  background: #FFCCCC;
  margin: 0;
  color: initial;
  padding: 16px 10px;
  font-size: 13px;
  border-radius: 4px;
  font-family: monospace;
}

<div
  class="c0"
>
  <div
    class="c1"
  >
    <svg
      aria-hidden="true"
      class="svg-inline--fa fa-cloud-showers-heavy fa-w-16 "
      data-icon="cloud-showers-heavy"
      data-prefix="fas"
      focusable="false"
      role="img"
      viewBox="0 0 512 512"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M183.9 370.1c-7.6-4.4-17.4-1.8-21.8 6l-64 112c-4.4 7.7-1.7 17.5 6 21.8 2.5 1.4 5.2 2.1 7.9 2.1 5.5 0 10.9-2.9 13.9-8.1l64-112c4.4-7.6 1.7-17.4-6-21.8zm96 0c-7.6-4.4-17.4-1.8-21.8 6l-64 112c-4.4 7.7-1.7 17.5 6 21.8 2.5 1.4 5.2 2.1 7.9 2.1 5.5 0 10.9-2.9 13.9-8.1l64-112c4.4-7.6 1.7-17.4-6-21.8zm-192 0c-7.6-4.4-17.4-1.8-21.8 6l-64 112c-4.4 7.7-1.7 17.5 6 21.8 2.5 1.4 5.2 2.1 7.9 2.1 5.5 0 10.9-2.9 13.9-8.1l64-112c4.4-7.6 1.7-17.4-6-21.8zm384 0c-7.6-4.4-17.4-1.8-21.8 6l-64 112c-4.4 7.7-1.7 17.5 6 21.8 2.5 1.4 5.2 2.1 7.9 2.1 5.5 0 10.9-2.9 13.9-8.1l64-112c4.4-7.6 1.7-17.4-6-21.8zm-96 0c-7.6-4.4-17.4-1.8-21.8 6l-64 112c-4.4 7.7-1.7 17.5 6 21.8 2.5 1.4 5.2 2.1 7.9 2.1 5.5 0 10.9-2.9 13.9-8.1l64-112c4.4-7.6 1.7-17.4-6-21.8zM416 128c-.6 0-1.1.2-1.6.2 1.1-5.2 1.6-10.6 1.6-16.2 0-44.2-35.8-80-80-80-24.6 0-46.3 11.3-61 28.8C256.4 24.8 219.3 0 176 0 114.2 0 64 50.1 64 112c0 7.3.8 14.3 2.1 21.2C27.8 145.8 0 181.5 0 224c0 53 43 96 96 96h320c53 0 96-43 96-96s-43-96-96-96z"
        fill="currentColor"
      />
    </svg>
  </div>
  <div
    class="c2"
  >
    Error
  </div>
  <div
    class="c3"
  >
    Sorry, we've encountered an issue
  </div>
  <div
    class="c4"
  >
    error details:
  </div>
  <div
    class="c5"
  >
    message
  </div>
</div>
`;

exports[`ErrorHandler Snapshot tests it matches the snapshot for mode singleLine 1`] = `
.c0 {
  margin: 1vh auto;
  background: #FFECEC;
  padding: 16px;
  width: 260px;
  border-radius: 6px;
  color: #9C2828;
}

.c1 {
  text-align: center;
  font-size: 24px;
  margin: 4px;
}

.c2 {
  text-align: center;
  font-size: 24px;
  font-weight: bold;
  margin-bottom: 16px;
}

.c3 {
  font-size: 16px;
  margin-bottom: 16px;
  text-align: center;
}

.c4 {
  font-size: 10px;
  font-weight: bold;
  text-align: center;
  margin-bottom: 4px;
}

.c5 {
  background: #FFCCCC;
  margin: 0;
  color: initial;
  padding: 16px 10px;
  font-size: 13px;
  border-radius: 4px;
  font-family: monospace;
}

<div
  class="c0"
>
  <div
    class="c1"
  >
    <svg
      aria-hidden="true"
      class="svg-inline--fa fa-cloud-showers-heavy fa-w-16 "
      data-icon="cloud-showers-heavy"
      data-prefix="fas"
      focusable="false"
      role="img"
      viewBox="0 0 512 512"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M183.9 370.1c-7.6-4.4-17.4-1.8-21.8 6l-64 112c-4.4 7.7-1.7 17.5 6 21.8 2.5 1.4 5.2 2.1 7.9 2.1 5.5 0 10.9-2.9 13.9-8.1l64-112c4.4-7.6 1.7-17.4-6-21.8zm96 0c-7.6-4.4-17.4-1.8-21.8 6l-64 112c-4.4 7.7-1.7 17.5 6 21.8 2.5 1.4 5.2 2.1 7.9 2.1 5.5 0 10.9-2.9 13.9-8.1l64-112c4.4-7.6 1.7-17.4-6-21.8zm-192 0c-7.6-4.4-17.4-1.8-21.8 6l-64 112c-4.4 7.7-1.7 17.5 6 21.8 2.5 1.4 5.2 2.1 7.9 2.1 5.5 0 10.9-2.9 13.9-8.1l64-112c4.4-7.6 1.7-17.4-6-21.8zm384 0c-7.6-4.4-17.4-1.8-21.8 6l-64 112c-4.4 7.7-1.7 17.5 6 21.8 2.5 1.4 5.2 2.1 7.9 2.1 5.5 0 10.9-2.9 13.9-8.1l64-112c4.4-7.6 1.7-17.4-6-21.8zm-96 0c-7.6-4.4-17.4-1.8-21.8 6l-64 112c-4.4 7.7-1.7 17.5 6 21.8 2.5 1.4 5.2 2.1 7.9 2.1 5.5 0 10.9-2.9 13.9-8.1l64-112c4.4-7.6 1.7-17.4-6-21.8zM416 128c-.6 0-1.1.2-1.6.2 1.1-5.2 1.6-10.6 1.6-16.2 0-44.2-35.8-80-80-80-24.6 0-46.3 11.3-61 28.8C256.4 24.8 219.3 0 176 0 114.2 0 64 50.1 64 112c0 7.3.8 14.3 2.1 21.2C27.8 145.8 0 181.5 0 224c0 53 43 96 96 96h320c53 0 96-43 96-96s-43-96-96-96z"
        fill="currentColor"
      />
    </svg>
  </div>
  <div
    class="c2"
  >
    Error
  </div>
  <div
    class="c3"
  >
    Sorry, we've encountered an issue
  </div>
  <div
    class="c4"
  >
    error details:
  </div>
  <div
    class="c5"
  >
    message
  </div>
</div>
`;

exports[`ErrorHandler Snapshot tests it matches the snapshot for mode tiny 1`] = `
.c0 {
  position: relative;
  display: inline-block;
  z-index: +1;
}

.c1 {
  cursor: pointer;
}

.c2 {
  background: #FFECEC;
  border-radius: 50%;
  padding: 4px;
  width: 34px;
  color: #9C2828;
  margin-left: 4px;
}

.c3 {
  text-align: center;
  font-size: 24px;
  margin: 4px;
}

<div
  class="c0"
>
  <div
    class="c1"
  >
    <div
      class="c2"
    >
      <div
        class="c3"
      >
        <svg
          aria-hidden="true"
          class="svg-inline--fa fa-cloud-showers-heavy fa-w-16 "
          data-icon="cloud-showers-heavy"
          data-prefix="fas"
          focusable="false"
          role="img"
          viewBox="0 0 512 512"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M183.9 370.1c-7.6-4.4-17.4-1.8-21.8 6l-64 112c-4.4 7.7-1.7 17.5 6 21.8 2.5 1.4 5.2 2.1 7.9 2.1 5.5 0 10.9-2.9 13.9-8.1l64-112c4.4-7.6 1.7-17.4-6-21.8zm96 0c-7.6-4.4-17.4-1.8-21.8 6l-64 112c-4.4 7.7-1.7 17.5 6 21.8 2.5 1.4 5.2 2.1 7.9 2.1 5.5 0 10.9-2.9 13.9-8.1l64-112c4.4-7.6 1.7-17.4-6-21.8zm-192 0c-7.6-4.4-17.4-1.8-21.8 6l-64 112c-4.4 7.7-1.7 17.5 6 21.8 2.5 1.4 5.2 2.1 7.9 2.1 5.5 0 10.9-2.9 13.9-8.1l64-112c4.4-7.6 1.7-17.4-6-21.8zm384 0c-7.6-4.4-17.4-1.8-21.8 6l-64 112c-4.4 7.7-1.7 17.5 6 21.8 2.5 1.4 5.2 2.1 7.9 2.1 5.5 0 10.9-2.9 13.9-8.1l64-112c4.4-7.6 1.7-17.4-6-21.8zm-96 0c-7.6-4.4-17.4-1.8-21.8 6l-64 112c-4.4 7.7-1.7 17.5 6 21.8 2.5 1.4 5.2 2.1 7.9 2.1 5.5 0 10.9-2.9 13.9-8.1l64-112c4.4-7.6 1.7-17.4-6-21.8zM416 128c-.6 0-1.1.2-1.6.2 1.1-5.2 1.6-10.6 1.6-16.2 0-44.2-35.8-80-80-80-24.6 0-46.3 11.3-61 28.8C256.4 24.8 219.3 0 176 0 114.2 0 64 50.1 64 112c0 7.3.8 14.3 2.1 21.2C27.8 145.8 0 181.5 0 224c0 53 43 96 96 96h320c53 0 96-43 96-96s-43-96-96-96z"
            fill="currentColor"
          />
        </svg>
      </div>
    </div>
  </div>
</div>
`;
